src/object/interrupt.c

```diff
@@ -145,12 +145,23 @@ void invokeIRQHandler_AckIRQ(irq_t irq)
 
 #if defined ENABLE_SMP_SUPPORT && defined CONFIG_ARCH_ARM
     if (IRQ_IS_PPI(irq) && IRQT_TO_CORE(irq) != getCurrentCPUIndex()) {
+#ifdef CONFIG_ARM_GIC_V3_SUPPORT
+        /* According to the GICv3 spec, SPIs can be deactivated from any PE,
+         * but SGIs and PPIs must be deactivated from their target PE.
+         */
+        doRemoteDeactivatePrivateInterrupt(IRQT_TO_CORE(irq), IRQT_TO_IDX(irq));
+#else /* CONFIG_ARM_GIC_V3_SUPPORT */
         doRemoteMaskPrivateInterrupt(IRQT_TO_CORE(irq), false, IRQT_TO_IDX(irq));
+#endif /* CONFIG_ARM_GIC_V3_SUPPORT */
         return;
     }
 #endif
-    maskInterrupt(false, irq);
-#endif
+    if (config_set(CONFIG_ARM_GIC_V3_SUPPORT)) {
+        deactivateInterrupt(irq);
+    } else {
+        maskInterrupt(false, irq);
+    }
+#endif /* CONFIG_ARCH_RISCV */
```

Comment:
This comment it also a bit confusing, as the block above is actually the `!CONFIG_ARCH_RISCV` part. Actually, it seems `invokeIRQHandler_AckIRQ` has become a very architectuere specific function now.

Reply:
I mainly try to make my kernel changes that are introducing new functionality have as little change to the AST of the code so that the proof changes for the new feature is minimized.

The irq layers of the kernel are starting to get a lot of stuff like this. It seems like it would be nicer to start cleaning it up, but maybe it'd be better to wait until more changes have come through for improving IRQ virtualization on aarch64 virtualization systems.